docs: warn about using different env variables for `runtimeConfig` (#24612)
nuxt/nuxt: docs/2.guide/3.going-further/10.runtime-config.md

})
```

> 

Reviewer: ```suggestion
```

1. Only a specially-named environment variable can override a runtime config property. That is, an uppercase environment variable starting with `NUXT_` which uses `_` to separate keys and case changes.

::callout{color="amber" icon="i-ph-warning-duotone"}

> Setting the default of `runtimeConfig` values to a *different environment variable* (for example setting `myVar` to `process.env.OTHER_VARIABLE`) will only work during build-time and will break on runtime.

Reviewer 2: ```suggestion
Setting the default of `runtimeConfig` values to *differently named environment variables* (for example setting `myVar` to `process.env.OTHER_VARIABLE`) will only work during build-time and will break on runtime.
```

> It is advised to use the envrionment variables that match the structure of your `runtimeConfig` object.

Reviewer 2: ```suggestion
It is advised to use environment variables that match the structure of your `runtimeConfig` object.
```

1. Only a specially-named environment variable can override a runtime config property. That is, an uppercase environment variable starting with `NUXT_` which uses `_` to separate keys and case changes.

::callout{color="amber" icon="i-ph-warning-duotone"}
Setting the default of `runtimeConfig` values to a *different environment variable* (for example setting `myVar` to `process.env.OTHER_VARIABLE`) will only work during build-time and will break on runtime.
It is advised to use the envrionment variables that match the structure of your `runtimeConfig` object.
::

#### Example

```sh [.env]
NUXT_API_SECRET=api_secret_token
NUXT_PUBLIC_API_BASE=https://nuxtjs.org

> SOME_OTHER_VAR=true

Reviewer: ```suggestion
```
